src/main/java/picard/sam/BuildBamIndex.java

```diff
@@ -88,7 +88,7 @@ protected int doWork() {
         // set default output file - input-file.bai
         if (OUTPUT == null) {
 
-            final String baseFileName = inputPath.getFileName().toString();
+            final String baseFileName = INPUT.hasFileSystemProvider() ? inputPath.toAbsolutePath().toString() : inputPath.getFileName().toString();
```

Comment:
The best fix for this would be to change `OUTPUT` to have the same type (`PicardHtsPath`) as `INPUT`, so that they have the same semantics. Keeping `OUTPUT` as `File` while `INPUT` is a `PicardHtsPath` will most likely cause problems with cloud inputs.

Reply:
Awesome, I was hoping you might have a better solution to this :)